zkvm/src/contract.rs

```diff
@@ -91,9 +78,45 @@ impl Input {
             txid,
         })
     }
+
+    // Creates a FrozenContract from payload (which is a vector of (qty, flv)) and pred.
+    // Serializes the contract, and uses the serialized contract and txid to generate a utxo.
+    // Returns an Input with the contract, txid, and utxo.
+    pub fn new(payload: Vec<(Commitment, Commitment)>, pred: Predicate, txid: TxID) -> Self {
```

Comment:
let the caller decide how to provide the payload:
```suggestion
    pub fn new<I>(payload: I, pred: Predicate, txid: TxID) -> Self 
    where I: IntoIterator<Item=(Commitment, Commitment)>
    {
```